client/packages/programs/src/VaccineCourseEditModal/VaccineCourseEditModal.tsx

```diff
@@ -81,6 +81,13 @@ interface VaccineCourseEditModalProps {
   mode: ModalMode | null;
 }
 
+function doseIndex(
+  doses: VaccineCourseDoseFragment[],
+  dose: VaccineCourseDoseFragment
+) {
+  return doses.indexOf(dose) + 1;
+}
```

Comment:
I am quite wary of list indexes relying on order of the array - though I see the repo layer does sort by min age

https://github.com/msupply-foundation/open-msupply/blob/aa859cbf1345acac1a8d336691d9d05464a4b68d/server/repository/src/db_diesel/vaccine_course/vaccine_course_dose.rs#L89-L90

Regardless of sorting or having an row ordering number field, could be some weird edge cases with soft deleted doses but I think it's fuzzy what to even do about them and unlikely to occur anyway.

Reply:
Yeah... I think this is okay though, we're not saving this number just using it for display. Yeah, if you have 5 doses, then delete the 4th one, it will say dose # 4, label = Dose 5... but at least dose # column and error message line up? 🤷‍♀️